lib/archethic/p2p/message.ex

```diff
@@ -1438,89 +1460,110 @@ defmodule Archethic.P2P.Message do
     %Ok{}
   end
 
-  def process(%ReplicateTransactionChain{
-        transaction: tx,
-        replying_node: replying_node_public_key
-      }) do
-    Task.Supervisor.start_child(TaskSupervisor, fn ->
-      response =
-        case Replication.validate_and_store_transaction_chain(tx) do
-          :ok ->
-            tx_summary = TransactionSummary.from_transaction(tx)
+  def process(
+        %ReplicateTransactionChain{
+          transaction: tx,
+          replying_node: replying_node_public_key
+        },
+        _
+      ) do
+    storage_nodes =
+      Election.chain_storage_nodes(
+        tx.address,
+        P2P.authorized_and_available_nodes()
+      )
 
-            %AcknowledgeStorage{
-              address: tx.address,
-              signature:
-                Crypto.sign_with_first_node_key(TransactionSummary.serialize(tx_summary)),
-              node_public_key: Crypto.first_node_public_key()
-            }
+    # Replicate transaction chain only if the current node is one of the chain storage nodes
+    if Utils.key_in_node_list?(storage_nodes, Crypto.first_node_public_key()) do
```

Comment:
You should not control the storage node pool if the transaction is a network type

Reply:
Why verify also the io storage nodes here ? For the IO it's checked on the %ReplicateTransaction{} message

Reply:
Because when we replicate a network transaction, the validations nodes send the message `ReplicateTransactionChain` to all storage and IO nodes

Reply:
For network transaction, ReplicateTransactionChain are sent to all the authorized nodes, and ReplicateTransaction is send to the all the rest of the nodes.

Reply:
```
context
    |> ValidationContext.get_io_replication_nodes()
    |> P2P.broadcast_message(
      if Transaction.network_type?(validated_tx.type),
        do: %ReplicateTransactionChain{
          transaction: validated_tx
        },
        else: %ReplicateTransaction{transaction: validated_tx}
    )
```

Reply:
Forgotten this one.
I guess the chain storage nodes are only the authorized to provide more security during the tx validation, then the chain is replicated on all the nodes.